### ISSUE TITLE
Materialization strategy `table` for seeds produces broken SQL

### ISSUE THREAD

**Issue by dejan** [databricks/dbt-databricks#113]
### Describe the bug
Materialization strategy `table` for seeds produces broken SQL.

```
/* {"app": "dbt", "dbt_version": "1.1.0", "profile_name": "z", "target_name": "dev", "node_id": "seed.z.channel_mapping"} */
    create or replace table desimic_zms.channel_mapping
    using delta
    location 's3://bucket/path/channel_mapping'
    as
      None
------^^^
```

### Steps To Reproduce

dbt_project.yml:

```
seeds:
  +location_root: "s3://bucket/path"
  +file_format: delta
  +materialization: table
``` 

I'm not sure what the default materialization for seed is (dbt documentation is not clear on this) , but I put `table` because by default with a combination of external tables, it appends records to the table, so I'm assuming it's `incremental` by default (surprisingly). 

### Expected behavior
This is debatable, but I expect seeds to behave like models with the materialization strategy `table`. 

### System information
**The output of `dbt --version`:**
```
Core:
  - installed: 1.1.0
  - latest:    1.1.1 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - spark:      1.1.0 - Up to date!
  - databricks: 1.1.0 - Up to date!
```

**The operating system you're using: OS X Big Sur**

**The output of `python --version`: Python 3.8.10**

**Comment by ueshin** [comment]
@dejan Thanks for reporting!
I'm afraid I think you are misunderstanding how to use [`seed`](https://docs.getdbt.com/reference/commands/seed) command.

> The `dbt seed` command will load `csv` files in the `seed-paths` directory of your dbt project.

The loaded data will be stored in a table named the same as the `csv` file.

I guess [`source`](https://docs.getdbt.com/docs/building-a-dbt-project/using-sources) is better fit to your case? 

@jtcohen6 Although I don't think `materialization` in the `seeds` config has effect, is there any chance to generate the weird query?
I tried to set `materialization` in the `seed` config in my local, but it seems working without any effect and any errors.

**Comment by jtcohen6** [comment]
Indeed, seeds want to be materialized as `'seed'`! That's the name of the materialization which a seed expects to use. Configuring it to materialize as a `'table'` instead will lead to some weird errors. It's incumbent on us to add better validation / a clearer error in that case: https://github.com/dbt-labs/dbt-core/issues/5229

Specifying `location_root` and `file_format` should work as expected.

**Comment by dejan** [comment]
@ueshin @jtcohen6 thanks for the quick response!

Sorry for being lazy to do a proper demo of the issue so we're better aligned with what the issue actually is. Here you go: https://github.com/dejan/dbt-demo-inconsistent-seeds

^ this is the original issue I encountered and I mentioned this in the first message as the default behavior for external tables. I assumed it might be for whatever reason expected behavior (that different materialization is used depending if `location_root` is used or not) so I attempted to configure the materialization for seed as a `table` and that's when I run into a broken SQL reported as a title of the issue.

After reading your messages I think the core issue is the inconsistent behavior of seeds depending on table type.

_As a side-note, also for consistency, it would be good if seeds share the materialization strategies with models. There might be some implementation details that justify having `seeds` as a strategy but if a seed can only become a table, why is that not also `table` materialization?_
